src/test/MapTest.cpp

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) 2005 - 2015 Settlers Freaks (sf-team at siedler25.org)
+//
+// This file is part of Return To The Roots.
+//
+// Return To The Roots is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 2 of the License, or
+// (at your option) any later version.
+//
+// Return To The Roots is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Return To The Roots. If not, see <http://www.gnu.org/licenses/>.
+
+#include "mapGenerator/Map.h"
+#include <boost/test/unit_test.hpp>
+
+BOOST_AUTO_TEST_SUITE(MapTest)
+
+/**
+ * Tests the default constructor of the Map class. Width and height of the map
+ * are expected to be zero.
+ */
+BOOST_FIXTURE_TEST_CASE(Constructor_DefaultZeroSize, Map)
+{
+    Map* map = new Map();
+    
+    BOOST_REQUIRE_EQUAL(map->width,  (unsigned)0);
+    BOOST_REQUIRE_EQUAL(map->height, (unsigned)0);
+    
+    delete map;
+}
+
+/**
+ * Tests the constructor of the Map class. The new map should have correct size 
+ * according to the constructor parameters.
+ */
+BOOST_FIXTURE_TEST_CASE(Constructor_CorrectSize, Map)
+{
+    Map* map = new Map(64, 64, "name", "author");
+
+    BOOST_REQUIRE_EQUAL(map->width,             (unsigned)64);
+    BOOST_REQUIRE_EQUAL(map->height,            (unsigned)64);
+    BOOST_REQUIRE_EQUAL(map->z.size(),          (unsigned)4096);
+    BOOST_REQUIRE_EQUAL(map->textureRsu.size(), (unsigned)4096);
+    BOOST_REQUIRE_EQUAL(map->textureLsd.size(), (unsigned)4096);
+    BOOST_REQUIRE_EQUAL(map->build.size(),      (unsigned)4096);
+    BOOST_REQUIRE_EQUAL(map->shading.size(),    (unsigned)4096);
+    BOOST_REQUIRE_EQUAL(map->resource.size(),   (unsigned)4096);
+    BOOST_REQUIRE_EQUAL(map->road.size(),       (unsigned)4096);
+    BOOST_REQUIRE_EQUAL(map->objectType.size(), (unsigned)4096);
+    BOOST_REQUIRE_EQUAL(map->objectInfo.size(), (unsigned)4096);
+    BOOST_REQUIRE_EQUAL(map->animal.size(),     (unsigned)4096);
+    BOOST_REQUIRE_EQUAL(map->unknown1.size(),   (unsigned)4096);
+    BOOST_REQUIRE_EQUAL(map->unknown2.size(),   (unsigned)4096);
+    BOOST_REQUIRE_EQUAL(map->unknown3.size(),   (unsigned)4096);
+    BOOST_REQUIRE_EQUAL(map->unknown5.size(),   (unsigned)4096);
+
+    delete map;
+}
+
+/**
+ * Tests the constructor of the Map class. The name must match the name specified
+ * in the constructor arguments.
+ */
+BOOST_FIXTURE_TEST_CASE(Constructor_CorrectName, Map)
+{
+    std::string name("name");
+    Map* map = new Map(64, 64, name, "author");
```

Comment:
Use instances, no pointers. And use at least 2 different authors. Like: MapA Author1, MapB Author2 After that require equal on each instance' author